### ISSUE TITLE
Add web links for pagination headers 

### ISSUE THREAD

**Issue by nonnib** [marshmallow-code/flask-smorest#133]
I have just started using the pagination support in flask-smorest and would really like to see links to pages in addition to the pagination information available in the X-Pagination header so that the client will not need to construct links themselves to paginate through results.

There is a standard called [RFC5988](https://tools.ietf.org/html/rfc5988
) that describes this and the [github api](https://developer.github.com/v3/guides/traversing-with-pagination/) uses that approach.

Basically, we would add a new response header that looks like this:

`GET https://www.mysite.com/pets/`

```
Link: <https://www.mysite.com/pets/?page=2>; rel="next", <https://www.mysite.com/pets/?page=20>; rel="last"
```

`GET https://www.mysite.com/pets/?page=3`
```
Link: <https://www.mysite.com/pets/>; rel="first", <https://www.mysite.com/pets/?page=2>; rel="prev", <https://www.mysite.com/pets/?page=20>; rel="last"
```

etc. Each `rel` is optional and is omitted if empty. For example, if pagination does not have a known number of pages The header might be:

`GET https://www.mysite.com/pets/`

```
Link: <https://www.mysite.com/pets/?page=2>; rel="next"
```

**Comment by lafrech** [comment]
I never really understood the point for HATEOAS since an API is made specifically to interact with software, not with a human. From a client developer POV, I prefer to get IDs and build URIs with them. I have the ID -> URI function somewhere in my code already. There must be a good reason people like to get links. I just don't get it. Maybe just a matter of personal preference.

I don't think I'd add this feature to the core, but I'm happy to provide hooks to achieve it.

The rework in #153 should help achieving it by overriding `_make_pagination_metadata`. Documenting it correctly is another story.

Anyone going forward with this is welcome to post his results here.

**Comment by nonnib** [comment]
That's a fair point and I realize that the `Link:` header rfc is not very nice to look at, but it is semi-standardized. For example, the python `requests` library ([link](https://requests.readthedocs.io/en/master/user/advanced/#link-headers)) knows how to navigate them which makes things easier for python client developers.

I personally think it's important to follow the industry expectations for these things rather than roll our own. I don't think I've seen the embedded json in header approach used successfully in api's, but I might be wrong.

I'll look at #153 for my specific usecase though. :)

**Comment by lafrech** [comment]
Yes. I don't do pagination in json payload either. I don't mean to support this in the code but I'd like to provide hooks allowing users to do it. That's the point of #153.

I meant that after #153 is merged, you should be able to achieve what you want by overriding `_make_pagination_metadata`.

**Comment by nonnib** [comment]
Yes, I understand. No worries.

**Comment by lafrech** [comment]
There may be other people interested in this, so please keep us posted if you manage to do it.

Perhaps I could add it as an example/test.
